src/core/model_accessing_API.jl

```diff
@@ -16,26 +18,29 @@ function add_agent_to_model!(agent::AbstractAgent, model::DictABM)
     maxid = getfield(model, :maxid)
     new_id = agent.id
     if maxid[] < new_id; maxid[] = new_id; end
-    extra_actions_after_add!(agent, model)
     return
 end
 
-function add_agent_to_model!(agent::AbstractAgent, model::VecABM)
+function add_agent_to_container!(agent::AbstractAgent, model::Union{VecABM, StructVecABM})
     agent.id != nagents(model) + 1 && error(lazy"Cannot add agent of ID $(agent.id) in a vector ABM of $(nagents(model)) agents. Expected ID == $(nagents(model)+1).")
     push!(agent_container(model), agent)
-    extra_actions_after_add!(agent, model)
     return
 end
 
-# This is extended for event based models (in their file)
+# This is extended for event based models
 extra_actions_after_add!(agent, model::StandardABM) = nothing
+function extra_actions_after_add!(agent, model::EventQueueABM{S,A,<:Union{AbstractDict, AbstractVector}} where {S,A})
+    getfield(model, :autogenerate_on_add) && add_event!(agent, model)
+end
+function extra_actions_after_add!(agent, model::EventQueueABM{S,A,<:StructVector} where {S,A})
+    getfield(model, :autogenerate_on_add) && add_event!(model[agent.id], model)
+end
 
 function remove_agent_from_model!(agent::AbstractAgent, model::DictABM)
     delete!(agent_container(model), agent.id)
     return
 end
-
-function remove_agent_from_model!(agent::AbstractAgent, model::VecABM)
+function remove_agent_from_model!(agent::AbstractAgent, model::Union{VecABM, StructVecABM})
```

Comment:
```suggestion
function remove_agent_from_container!(agent::AbstractAgent, model::Union{VecABM, StructVecABM})
```

for consistency this should be renamed